tests/conftest.py

```diff
@@ -131,7 +131,11 @@ def xp(
         # suppress unused-ignore to run mypy in -e lint as well as -e dev
         jax.config.update("jax_enable_x64", True)  # type: ignore[no-untyped-call,unused-ignore]
 
-    return xp
+    device = None
+    if library == Backend.ARRAY_API_STRICT_DEVICE1:
+        import array_api_strict
+        device = array_api_strict.Device("device1")
+    return xp, device
```

Comment:
I was trying to add multi-device testing here similar to the way scikit-learn does it.
https://github.com/scikit-learn/scikit-learn/pull/30090/files

I don't know if this is the right approach, though.
I didn't fix all the tests - I wanted to get some feedback before making a very invasive change like this (I'd need to migrate all the tests).